### PR TITLE
Reset gateway when valid gateway URL is submitted in settings

### LINKED ISSUE #408: Stale Gateway URL is still used for sync after update
Ran into this while dogfooding sync.

# Reproduction 
- Fresh install of the app
- Complete initial setup
- Open settings
- Update gateway URL
- Hit "Sync with Gateway"
- Sync fails
- Logs show the old Gateway URL being used

image

```
[app] Syncing with gateway: http://127.0.0.1:4433
```

# Expected Behaviour
Use the new URL for sync immediately after updating the setting field

# Workaround
Relaunching the app loads the updated Gateway URL

### PR DESCRIPTION
**Pull request by gordonbrander**
Fixes #408 

Changes:

- Call `NoosphereService.resetGateway()` when a valid gateway URL is submitted in settings
- Configure gateway TextField for URLs
    - Autocapitalize off
    - Autocomplete off
    - URL keyboard
- Introduce URL.isHTTP() extension, with tests.
- Limit labeled content in settings (including gateway URL field) to one line

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/AppView.swift [hunk 1]
```diff
@@ -448,18 +444,30 @@ struct AppModel: ModelProtocol {
...
+            return Update(state: fallback)
         }
+
         var model = state
         model.gatewayURL = gatewayURL
         model.gatewayURLTextField = gatewayURL
         model.isGatewayURLTextFieldValid = true
+
+        // Persist to UserDefaults
         AppDefaults.standard.gatewayURL = gatewayURL
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/AppView.swift:467]: Similar to my clipboard comment the other day, I find myself wondering if `AppDefaults` should also be available via the `AppEnvironment`

**Review by bfollington — APPROVED** [review]
LGTM, tested and confirmed the fix

File: xcode/Subconscious/Shared/Components/AppView.swift (hunk 1 above)
Comment (gordonbrander) [reply xcode/Subconscious/Shared/Components/AppView.swift:467]: Fair point. Xcode does expose UserDefaults for unit testing, so we don't need to mock. However, there is a good argument to be made to make this part of environment anyway.

AppDefaults is used in several places. This change would touch a number of other files, so I suggest following up in another PR.

Comment (gordonbrander) [reply xcode/Subconscious/Shared/Components/AppView.swift:467]: https://github.com/subconsciousnetwork/subconscious/issues/413

**Pull request merged**